### DISCUSSION TITLE
Is it possible to change Foliate's TTS client?

### DISCUSSION THREAD

**Discussion by ebulutt** [johnfactotum/foliate#752] [Q&A]
**Question:**
I am not a fan of reading books from a computer screen.

I install Foliate to listen to my epub books while I am working on the PC. It uses **festival tts** by default.

I have **Glate** on **Ubuntu 20.04**. And they say that Glate has **Google TTS** included. Can I and how can I use Google TTS on Foliate instead of festival tts?

Thank you in advance!

**Version:**
 - Foliate version: 2.6.3
 - OS/Distribution and version: Ubuntu 20.04
- DE - Gnome 3
 - Installation method: I don't remember. Maybe snap?

**Comment by johnfactotum** [comment]
See the FAQ: https://github.com/johnfactotum/foliate/wiki#how-to-use-text-to-speech

**Comment by ebulutt** [comment]
(use man espeak-ng or man festival to see all the options)
what options and where? because I have tried both "man" but nothing has changed. I want to listen in Turkish language without a robotic voice.

**Comment by johnfactotum** [comment]
> (use man espeak-ng or man festival to see all the options) what options and where? because I have tried both "man" but nothing has changed.

You can run `man espeak-ng`, etc., in your terminal emulator to see the available command line options. Then you can specify those options in Foliate, if desired.

> I have **Glate** on **Ubuntu 20.04**. And they say that Glate has **Google TTS** included. Can I and how can I use Google TTS on Foliate instead of festival tts?

I think Glate is a GUI program. If you'd like to use Google's TTS with Foliate, I'd recommend using [gTTS](https://gtts.readthedocs.io/). Once gTTS is installed, save the following script (as described in the FAQ) to any location, say `/home/user/gtts.sh`:
```bash
#!/bin/bash
gtts-cli -l $FOLIATE_TTS_LANG_LOWER --file /dev/stdin | play -t mp3 - &
trap 'kill $!; exit 0' INT
wait
```
Then, in Foliate, set `/home/user/gtts.sh` as the TTS command.

**Reply by adityatelange** [reply]
Thanks a lot for this :+1:

**Comment by archisman-panigrahi** [comment]
In Ubuntu, in addition to `gtts`, one may have to install the packages `sox` and `libsox-fmt-mp3` for this script to work.

```
pip3 install --user gTTs
sudo apt-get install libsox-fmt-mp3 sox
```

**Comment by ebulutt** [comment]
Now I am almost there. I installed gtts (since there was some problem with my pip command I solve it with [this](https://stackoverflow.com/a/55931701/15823929).) Then install that libsox as @apandada1 said. Now, where and how I will "save" that script? Did you mean in .sh file on text editor? Sorry if I am silly. :)

**Comment by johnfactotum** [comment]
Yeah, pasting it into a text editor and saving the file should work.

**Comment by ebulutt** [comment]
I did everything you've said and preferences look like that now:
https://ibb.co/VCzhXZV
and no sound on test or book.

**Comment by ebulutt** [comment]
ops sorry user shouldn't be folder name but mine :)
but still don't work.

**Comment by archisman-panigrahi** [comment]
Try marking the script as executable.

`chmod +x /home/user/gtts.sh`

The script works for me in KDE Neon 20.04. It is hard to debug why it is not working for you. 

First check if gtts works in terminal.
Does this command say `hello`?

`gtts-cli 'hello' | play -t mp3 -`

If not, does it produce any error?

**Comment by ebulutt** [comment]
@apandada1 yes it says hello with that command. 
after chmod command there is no action. nothing.

**Comment by archisman-panigrahi** [comment]
Run foliate from the Terminal, and check the output when you run a text to speech in it.

**Comment by varlesh** [comment]
**sox** not good worked for me and not play first line.
But popular players worked fine. Try it!
**cvlc**:
```
gtts-cli -l $FOLIATE_TTS_LANG_LOWER --file /dev/stdin | cvlc --play-and-exit - &
trap 'kill $!; exit 0' INT
wait
```
**mpv**:
```
gtts-cli -l $FOLIATE_TTS_LANG_LOWER --file /dev/stdin | mpv --really-quiet - &
trap 'kill $!; exit 0' INT
wait
```

**Comment by code-ink-space** [comment]
In my case (probably similar to @ebulutt's problem of hearing no sound) I had to change the way I launch Foliate. If you're using .desktop application shortcuts, you may need to change the Exec value to make that wrapper tts script Foliate calls work right.
```
Exec=bash -c /usr/bin/foliate
```
I figured this thanks to @apandada1's suggestion to look at the verbose output from the terminal.

**Comment by twkonefal** [comment]
I installed Pied (https://github.com/Elleo/pied) which allowed me to easily preview, download, and select new voices for speech-dispatcher. Foliate then used this for TTS. Works great.

**Reply by archisman-panigrahi** [reply]
Can you edit to add more details about how to use it with the latest version of Foliate?

**Reply by sykul** [reply]
It's as simple as installing Pied, and then installing Piper inside the Pied GUI, then clicking on the voice you want. 
https://www.kentoseth.com/posts/2025/feb/10/how-to-get-natural-sounding-tts-text-to-speech-on-linux-okular-and-other-apps/
